src/blockchain_db/sqlite/db_sqlite.cpp

```diff
@@ -426,22 +461,55 @@ static std::pair<uint64_t, uint64_t> get_accrued_rewards_impl(
         WHERE address = ?
     )",
             address);
-    auto result = std::make_pair(height, static_cast<uint64_t>(rewards.value_or(0) / 1000));
-    return result;
+    return static_cast<uint64_t>(rewards.value_or(0) / 1000);
+}
+
+static std::optional<uint64_t> get_accrued_rewards_at_impl(
+        BlockchainSQLite& db, const std::string& address, uint64_t at_height) {
+    log::trace(logcat, "BlockchainDB_SQLITE {} for {}", __func__, address);
+    auto rewards = db.prepared_maybe_get<int64_t>(
+            R"(
+        SELECT amount
+        FROM batched_payments_accrued_recent
+        WHERE address = ? AND height = ?
+    )",
+            address,
+            static_cast<int64_t>(at_height));
+    if (!rewards) {
+        // No rewards found; check to see if we actually have any recent records for that height and
+        // if not, return a "don't know" nullopt value.  Otherwise we fall through and return an
+        // authoritive 0 value.
```

Comment:
Why 0? In the bls aggregator usage code that 0 is going to trigger the "invalid request because you have earnt 0 SENT" which is going to be misleading especially if you're targeting the tip of the chain after you've just received a block.

I think more appropriate is returning another std::nullopt, or, pull the maximum known rewards value (which would be the most correct output if you pass in a much higher height than the DB has)

Reply:
Good point; pushed an update.  (We don't have to actually need to query that one as we have a `height` member already tracking it).

Reply:
For heights that are in between min-height-we-have and curr-height we still do want to return 0, because that does indeed mean that address has no SENT earned.